### PR TITLE
Issue #6728: Constant Windowed Aggregation

### PR DESCRIPTION
**Pull request by hawkfish**
Add a new windowed aggregate accelerator that handles the case when all the aggregate values are the same for a partition. In this situation, we can just aggregate the values as then are scanned and copy the results out during computation.

The included benchmark is about 15x faster with this change.

### CODE REVIEW

File: src/execution/window_segment_tree.cpp [hunk 1]
```diff
@@ -6,6 +6,178 @@
...
+
+			if (nsel != inputs.size()) {
+				inputs.Slice(sel, nsel);
+			}
+		}
+
+		//	Aggregate the filtered rows into a single state
+		const auto count = inputs.size();
+		if (aggregate.simple_update) {
+			aggregate.simple_update(inputs.data.data(), aggr_input_data, inputs.ColumnCount(), state.data(), count);
```
Comment (Mytherin) [inline src/execution/window_segment_tree.cpp:147]: Could we add some extra tests for constant window aggregates? I wonder how well some of these edge cases are covered by our current set of tests. e.g.:

* Mix of constant/non-constant aggregate (`sum` and `lead` over same `OVER` clause)
* Constant aggregate with non-`simple_update`
* Constant aggregate with not-nicely-aligned partitions (e.g. partitions are 73, 75, 77, 79, 81, 83, ... rows)
* Constant aggregate with large partitions (> vector size), also not nicely-aligned

**Review by Mytherin — COMMENTED** [review]
Thanks for the PR! Impressive results. Some comments:

File: src/execution/window_segment_tree.cpp (hunk 1 above)
Comment (hawkfish) [reply src/execution/window_segment_tree.cpp:147]: The benchmark has 3000 element partitions, and the row count is low enough that the hash groups have multiple partitions. But I can certainly add the mixtures.

**Comment by Mytherin** [comment]
Thanks!

**Pull request merged**